### PR TITLE
KOGITO-1802: Create new VSCode Extensions distributions

### PR DESCRIPTION
**Pull request by tiagobento**
On this PR:
- Creation of GitHub Action `ci.yml` to build and upload the artifacts for every PR.
- Creation of four VS Code extension distributions:
  1. BPMN Editor
  2. DMN Editor
  3. Kogito Bundle
  4. Red Hat Business Automation Bundle

### CODE REVIEW

**Review by ederign — APPROVED** [review]

File: lerna.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "version": "0.4.2",
```
Comment (tiagobento) [inline lerna.json:2]: We should discuss what's the best first version for these packages.

File: packages/kie-bc-editors-unpacked/pom.xml [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                <enabled>true</enabled>
+                <updatePolicy>always</updatePolicy>
+            </snapshots>
+        </repository>
+    </repositories>
+
+    <properties>
+        <version.dmn.webapp>7.38.0-SNAPSHOT</version.dmn.webapp>
+        <version.bpmn.webapp>7.38.0-SNAPSHOT</version.bpmn.webapp>
+        <version.scesim.webapp>7.38.0-SNAPSHOT</version.scesim.webapp>
```
Comment (tiagobento) [inline packages/kie-bc-editors-unpacked/pom.xml:42]: We should discuss what's the best first version for the editors.

File: packages/vscode-extension-bpmn-editor/package.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    "build-ext": "rm -rf dist && webpack --mode production --devtool none",
+    "pack-ext": "vsce package --yarn -o ./dist/vscode_extension_bpmn_editor_$npm_package_version.vsix",
+    "pack": "yarn build-ext && yarn pack-ext",
+    "compile": "yarn build-ext",
+    "watch": "yarn build-ext"
+  },
+  "dependencies": {
+    "@kogito-tooling/kie-bc-editors": "0.4.2",
+    "@kogito-tooling/microeditor-envelope": "0.4.2",
+    "@kogito-tooling/vscode-extension": "0.4.2",
```
Comment (tiagobento) [inline packages/vscode-extension-bpmn-editor/package.json:20]: We should discuss what's the best first version for the `@kogito-tooling/*` packages.

File: packages/vscode-extension-dmn-editor/package.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    "build-ext": "rm -rf dist && webpack --mode production --devtool none",
+    "pack-ext": "vsce package --yarn -o ./dist/vscode_extension_dmn_editor_$npm_package_version.vsix",
+    "pack": "yarn build-ext && yarn pack-ext",
+    "compile": "yarn build-ext",
+    "watch": "yarn build-ext"
+  },
+  "dependencies": {
+    "@kogito-tooling/kie-bc-editors": "0.4.2",
+    "@kogito-tooling/microeditor-envelope": "0.4.2",
+    "@kogito-tooling/vscode-extension": "0.4.2",
```
Comment (tiagobento) [inline packages/vscode-extension-dmn-editor/package.json:20]: We should discuss what's the best first version for the `@kogito-tooling/*` packages.

**Review by tiagobento — COMMENTED** [review]
@ederign I left some comments in line for us to discuss what we should do with the versions.

File: packages/kie-bc-editors-unpacked/pom.xml (hunk 2 above)
Comment (ederign) [reply packages/kie-bc-editors-unpacked/pom.xml:42]: For master, I think it's fine to keep the same as the community. As soon as we have a position for productization, we will need to branch/tag something and we change those properties accordingly.

File: lerna.json (hunk 1 above)
Comment (ederign) [reply lerna.json:2]: For master, I think it's fine to keep the same as the community. As soon as we have a position for productization, we will need to branch/tag something and we change those properties accordingly.

File: packages/vscode-extension-bpmn-editor/package.json (hunk 3 above)
Comment (ederign) [reply packages/vscode-extension-bpmn-editor/package.json:20]: For master, I think it's fine to keep the same as the community. As soon as we have a position for productization, we will need to branch/tag something and we change those properties accordingly.

File: packages/vscode-extension-dmn-editor/package.json (hunk 4 above)
Comment (ederign) [reply packages/vscode-extension-dmn-editor/package.json:20]: For master, I think it's fine to keep the same as the community. As soon as we have a position for productization, we will need to branch/tag something and we change those properties accordingly.

Comment (ederign) [reply packages/vscode-extension-dmn-editor/package.json:20]: /cc @porcelli

File: packages/vscode-extension-bpmn-editor/package.json (hunk 3 above)
Comment (ederign) [reply packages/vscode-extension-bpmn-editor/package.json:20]: /cc @porcelli

File: packages/kie-bc-editors-unpacked/pom.xml (hunk 2 above)
Comment (ederign) [reply packages/kie-bc-editors-unpacked/pom.xml:42]: /cc @porcelli

File: lerna.json (hunk 1 above)
Comment (ederign) [reply lerna.json:2]: /cc @porcelli

**Pull request merged**